### PR TITLE
Get locators update

### PR DESCRIPTION
**Pull request by alicevhenshaw**
## Description

PR for https://github.com/airswap/airswap-protocols/issues/224.

## Changes

- `getLocators` in `Indexer` and `Index` to return scores, locators and the next user's address. Previously it only returned locators.
- Additionally, the upper bound of array size in `Index.getLocators` has been tightened

## Tests
All passing

### CODE REVIEW

File: source/index/contracts/Index.sol [hunk 1]
```diff
@@ -148,13 +148,18 @@ contract Index is Ownable {
...
     */
   function getLocators(
     address start,
     uint256 count
-  ) external view returns (bytes32[] memory result) {
-
+  ) external view returns (
+    bytes32[] memory locators,
+    uint256[] memory scores,
+    address nextIdentifier
```
Comment (poiresel) [inline source/index/contracts/Index.sol]: should nextIdentifier be the name of the return term in line 189?

Comment (alicevhenshaw) [reply source/index/contracts/Index.sol]: I thought `nextIdentifier` was more descriptive of what was being returned to the called than `identifier`.

I could call it `nextIdentifier` throughout the function though

Comment (poiresel) [reply source/index/contracts/Index.sol]: yes, and i think that's accurate in terms how that variable is called. the initialized value is `entries[HEAD].next;`

**Comment by alicevhenshaw** [comment]
@poiresel Index.getScore also exists so I think that would be superfluous. Though we could combine them to one. Thoughts?

**Comment by poiresel** [comment]
> @poiresel Index.getScore also exists so I think that would be superfluous. Though we could combine them to one. Thoughts?

That works since they are both quick retrievals.

File: source/index/contracts/Index.sol [hunk 2]
```diff
@@ -11,7 +11,7 @@
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
-  limitations under the License.
+  sizeations under the License.
```
Comment (poiresel) [inline source/index/contracts/Index.sol]: bad copy and replace?

Comment (alicevhenshaw) [reply source/index/contracts/Index.sol]: hahahah yes. Will revert it.

File: source/index/contracts/Index.sol [hunk 3]
```diff
@@ -147,33 +147,47 @@ contract Index is Ownable {
     * @notice Get a Range of Locators
     * @dev start value of 0x0 starts at the head
     * @param start address Identifier to start with
-    * @param count uint256 Number of locators to return
-    * @return bytes32[] result List of locators
+    * @param limit uint256 Number of locators to return
+    * @return bytes32[] List of locators
+    * @return uint256[] List of scores corresponding to locators
+    * @return address The next identifier to provide for pagination
```
Comment (poiresel) [inline source/index/contracts/Index.sol]: next cursor? not sure if that's the right usage of the term but replace with something that's not identifier.

File: utils/order-utils/src/constants.js [hunk 4]
```diff
@@ -20,6 +20,9 @@ module.exports = {
   SECONDS_IN_DAY: 86400,
   EMPTY_ADDRESS: '0x0000000000000000000000000000000000000000',
   HEAD: '0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF',
```
Comment (poiresel) [inline utils/order-utils/src/constants.js]: why are some of the f's lower case and some upper case?

File: source/indexer/test/Indexer.js [hunk 5]
```diff
@@ -495,7 +512,7 @@ contract('Indexer', async ([ownerAddress, aliceAddress, bobAddress]) => {
     })
 
     it('Bob ensures there are no more intents the Indexer', async () => {
-      const intents = await indexer.getLocators.call(
+      result = await indexer.getLocators.call(
         tokenWETH.address,
         tokenDAI.address,
         EMPTY_ADDRESS,
```
Comment (poiresel) [inline source/indexer/test/Indexer.js]: can you make 10 a constant like DEFAULT_QUERY_SIZE or something?

File: utils/order-utils/src/constants.js (hunk 4 above)
Comment (alicevhenshaw) [reply utils/order-utils/src/constants.js]: Its a check summed Ethereum address

File: source/index/contracts/Index.sol [hunk 6]
```diff
@@ -146,34 +146,49 @@ contract Index is Ownable {
   /**
     * @notice Get a Range of Locators
     * @dev start value of 0x0 starts at the head
-    * @param start address Identifier to start with
-    * @param count uint256 Number of locators to return
-    * @return bytes32[] result List of locators
+    * @param cursor address Cursor to start with
+    * @param limit uint256 Number of locators to return
```
Comment (dmosites) [inline source/index/contracts/Index.sol]: *Maximum number of locators to return

File: source/index/contracts/Index.sol [hunk 7]
```diff
@@ -146,34 +146,49 @@ contract Index is Ownable {
...
+    bytes32[] memory locators,
+    uint256[] memory scores,
+    address nextCursor
+  ) {
+    address identifier;
+
+    // If a valid cursor is provided, start there.
+    if (cursor != address(0) && cursor != HEAD) {
+      // Check that the provided cursor exists.
+      require(_hasEntry(cursor), "START_ENTRY_NOT_FOUND");
```
Comment (dmosites) [inline source/index/contracts/Index.sol]: `CURSOR_NOT_FOUND`

File: source/indexer/test/Indexer-unit.js [hunk 8]
```diff
@@ -586,17 +594,19 @@ contract('Indexer Unit Tests', async accounts => {
   })
 
   describe('Test getLocators', async () => {
-    it('should return an empty array if the index doesnt exist', async () => {
-      let intents = await indexer.getLocators.call(
+    it('should return blank results if the index doesnt exist', async () => {
+      result = await indexer.getLocators.call(
```
Comment (ejwessel) [inline source/indexer/test/Indexer-unit.js]: Is the global supposed to be the `let result` here?

File: source/indexer/test/Indexer.js [hunk 9]
```diff
@@ -30,6 +36,8 @@ contract('Indexer', async ([ownerAddress, aliceAddress, bobAddress]) => {
   let tokenDAI
   let tokenWETH
 
+  let result
```
Comment (ejwessel) [inline source/indexer/test/Indexer.js]: global question again.

**Comment by ejwessel** [comment]
Other than my comments on result. This looks great.

File: source/index/test/Index-unit.js [hunk 10]
```diff
@@ -23,6 +29,8 @@ contract('Index Unit Tests', async accounts => {
   let snapshotId
   let index
 
+  let result
```
[the remark this answers is not in the document]
Comment (alicevhenshaw) [reply source/index/test/Index-unit.js]: just global because it got used so many times in so many tests

**Pull request merged**